sdk/keyvault/keyvault-certificates/src/certificatesModels.ts

```diff
@@ -624,3 +624,40 @@ export interface RecoverDeletedCertificateOptions extends coreHttp.OperationOpti
  * An interface representing optional parameters for {@link restoreCertificateBackup}.
  */
 export interface RestoreCertificateBackupOptions extends coreHttp.OperationOptions {}
+
+/**
+ * The contact information for the vault certificates.
+ */
+export interface CertificateContactAll {
+  /**
+   * Email address.
+   */
+  emailAddress: string;
+  /**
+   * Name.
+   */
+  name: string;
+  /**
+   * Phone number.
+   */
+  phone: string;
+}
+export type RequireAtLeastOne<T> = {
+  [K in keyof T]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<keyof T, K>>>;
```

Comment:
This looks fancy. For future us, can you comment how this works?

Reply:
For sure!

Reply:
Yeah, my understanding of this fanciness is a little fuzzy still too and I was the one who tweaked it.

What I get out of it:

`[keyof T]` at the bottom loops over each key property and makes a union out of a transformation to the base shape. 

This transformation makes one key required `Required<Pick<T, K>>` and the rest of the keys optional `Partial<Pick<T, Exclude<keyof T, K>>>`

What's weird about it is we're not applying a constant transformation to the interface, but doing one transformation per key and then union-ing them all back together.

Reply:
I'm asking Daniel Rosenwasser about what would be the appropriate wording for this.

Reply:
Daniel Rosenwasser is out of the office. I have updated this PR with what I believe is a reasonable explanation of the new type.